plugins/c8y_firmware_plugin/src/tests.rs

```diff
@@ -0,0 +1,775 @@
+            Ok(downloaded_path)
+        });
+
+    start_firmware_manager(
+        &mut tmp_dir,
+        broker.port,
+        c8y_http_client,
+        DEFAULT_REQUEST_TIMEOUT_SEC,
+    )
+    .await?;
+
+    // Mock download endpoint for the plugin to download a firmware from the cloud
+    let mock_http_server_host = mockito::server_url();
+    let cloud_firmware_url = format!("{mock_http_server_host}/some/cloud/url");
+
+    // Subscribe SmartREST endpoint for child device
+    let mut smartrest_messages = broker
+        .messages_published_on(format!("c8y/s/us/{CHILD_DEVICE_ID}").as_str())
+        .await;
+
+    // Publish a c8y_Firmware operation to the plugin
+    broker
+        .publish(
+            "c8y/s/ds",
+            format!(
+                "515,{CHILD_DEVICE_ID},{FIRMWARE_NAME},{FIRMWARE_VERSION},{cloud_firmware_url}"
+            )
+            .as_str(),
+        )
+        .await?;
+
+    // Assert the c8y_Firmware operation status mapping to EXECUTING(501) and FAILED(502)
+    mqtt_tests::assert_received_all_expected(
+        &mut smartrest_messages,
+        TEST_TIMEOUT_MS,
+        &["501,c8y_Firmware", "502,c8y_Firmware"],
+    )
+    .await;
+
+    Ok(())
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+#[serial_test::serial]
+async fn handle_request_timeout_child_device() -> anyhow::Result<()> {
+    let mut tmp_dir = TempTedgeDir::new();
+    create_required_directories(&mut tmp_dir);
+
+    let broker = mqtt_tests::test_mqtt_broker();
+    let mut c8y_http_client = MockC8YHttpProxy::new();
+    c8y_http_client
+        .expect_download_file()
+        .with(
+            predicate::always(),
+            predicate::always(),
+            predicate::always(),
+        )
+        .returning(|_, file_name, tmp_dir_path| {
+            let downloaded_path = tmp_dir_path.join(file_name);
+            std::fs::File::create(&downloaded_path)?;
+            Ok(downloaded_path)
+        });
+
+    start_firmware_manager(
+        &mut tmp_dir,
+        broker.port,
+        c8y_http_client,
+        Duration::from_secs(1),
+    )
+    .await?;
+
+    // Mock download endpoint for the plugin to download a firmware from the cloud
+    let mock_http_server_host = mockito::server_url();
+    let cloud_firmware_url = format!("{mock_http_server_host}/some/cloud/url");
+
+    // Subscribe SmartREST endpoint for child device
+    let mut smartrest_messages = broker
+        .messages_published_on(format!("c8y/s/us/{CHILD_DEVICE_ID}").as_str())
+        .await;
+
+    // Publish a c8y_Firmware operation to the plugin
+    broker
+        .publish(
+            "c8y/s/ds",
+            format!(
+                "515,{CHILD_DEVICE_ID},{FIRMWARE_NAME},{FIRMWARE_VERSION},{cloud_firmware_url}"
+            )
+            .as_str(),
+        )
+        .await?;
+
+    // Assert the c8y_Firmware operation status mapping to EXECUTING(501) and FAILED(502)
+    mqtt_tests::assert_received_all_expected(
```

Comment:
Don't you need to make sure that the timeout period has elapsed before doing this assertion? With a `sleep` may be.

Reply:
The firmware response timeout is 1 sec. We wait 5 seconds to check if all expected message are received inside assert. I don't think we need sleep.